### ISSUE TITLE
Can't get cudnn.BatchNormalization to work

### ISSUE THREAD

**Issue by vintermann** [soumith/cudnn.torch#281]
I get this error when trying to create a cudnn network with batch normalization:

```
/home/harald/torch/install/bin/luajit: /home/harald/torch/install/share/lua/5.1/nn/Container.lua:67: 
In 3 module of nn.Sequential:
...torch/install/share/lua/5.1/cudnn/BatchNormalization.lua:44: assertion failed!
stack traceback:
	[C]: in function 'assert'
	...torch/install/share/lua/5.1/cudnn/BatchNormalization.lua:44: in function 'createIODescriptors'
	...torch/install/share/lua/5.1/cudnn/BatchNormalization.lua:60: in function <...torch/install/share/lua/5.1/cudnn/BatchNormalization.lua:59>

```

Here's an example I've minimized. This fails if USE_CUDNN is true, works otherwise. 

```
require 'nn'

USE_CUDNN = true

net = nn.Sequential()
net:add(nn.SpatialConvolution(1,3,3,3,1,1,1,1))
net:add(nn.ReLU(true))
net:add(nn.SpatialBatchNormalization(3))

img = torch.Tensor(1,9,9)

if USE_CUDNN then
   require 'cunn'
   require 'cudnn'
   cudnn.convert(net, cudnn)
   net:cuda()
   img = img:cuda()
end

net:evaluate()
result = net:forward(img)
```

If I, instead of using cudnn.convert, manually replace them with their cudnn equivalents, it also fails. However, if I let SpatialBatchNormalization be nn while the other two are cudnn, it works.

I just updated torch, nn, cutorch, cunn, cudnn in that order. I use cuda 7.5.

**Comment by fmassa** [comment]
You need to feed a batch (4d) data.
try changing your input to something like `torch.random(2,1,9,9)` and it should work

**Comment by vintermann** [comment]
Thanks!
